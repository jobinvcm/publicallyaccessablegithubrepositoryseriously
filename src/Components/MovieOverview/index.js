import React from "react";
import Paper from "@material-ui/core/Paper";
import propTypes from 'prop-types';

const MovieOverview = (props) => {
  return (
    <div>
      <Paper elevation={2}>
        <img src={props.item.imageUrl} />
      </Paper>

      <div>{props.item.title}</div>
      <div>{props.item.year}</div>

    </div>
  );
};

MovieOverview.propTypes = {
  item: propTypes.shape({
    imageUrl: propTypes.string,
    title: propTypes.string,
    year: propTypes.string
  })
};

export default MovieOverview;
