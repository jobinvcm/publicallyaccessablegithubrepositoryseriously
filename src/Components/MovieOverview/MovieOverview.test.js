import React from 'react';
import { shallow } from 'enzyme';
import MovieOverview from '.';
import { isMainThread } from 'worker_threads';
import Paper from '@material-ui/core/Paper';
import { ExpansionPanelActions } from '@material-ui/core';

const mockItem = {
  title: "test title",
  year: "December 2016",
  imageUrl: "http://testimage"
}

describe('<MovieOverview />', () => {
  it('renders paper component', () => {
    const wrapper = shallow(<MovieOverview item={mockItem}/>);
    expect(wrapper).toMatchSnapshot();
  })

  it('should have page elemnt', () => {
    const wrapper =shallow(<MovieOverview item={mockItem} />);
    expect(wrapper.find(Paper)).toHaveLength(1);
  })
})