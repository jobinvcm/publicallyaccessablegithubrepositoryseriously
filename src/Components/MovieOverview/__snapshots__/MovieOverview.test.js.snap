// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieOverview /> renders paper component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MovieOverview
    item={
      Object {
        "imageUrl": "http://testimage",
        "title": "test title",
        "year": "December 2016",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <WithStyles(Paper)
          elevation={2}
        >
          <img
            src="http://testimage"
          />
        </WithStyles(Paper)>,
        <div>
          test title
        </div>,
        <div>
          December 2016
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <img
            src="http://testimage"
          />,
          "elevation": 2,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "src": "http://testimage",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "test title",
        },
        "ref": null,
        "rendered": "test title",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "December 2016",
        },
        "ref": null,
        "rendered": "December 2016",
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <WithStyles(Paper)
            elevation={2}
          >
            <img
              src="http://testimage"
            />
          </WithStyles(Paper)>,
          <div>
            test title
          </div>,
          <div>
            December 2016
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <img
              src="http://testimage"
            />,
            "elevation": 2,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "src": "http://testimage",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "test title",
          },
          "ref": null,
          "rendered": "test title",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "December 2016",
          },
          "ref": null,
          "rendered": "December 2016",
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
